lib/knapsack_pro/base_allocator_builder.rb

```diff
@@ -38,7 +38,29 @@ def test_file_pattern
     end
 
     def test_files
-      KnapsackPro::TestFileFinder.call(test_file_pattern)
+      found_test_files = KnapsackPro::TestFileFinder.call(test_file_pattern)
+
+      if adapter_class == KnapsackPro::Adapters::RSpecAdapter && KnapsackPro::Config::Env.rspec_split_by_test_examples?
+        test_files_count = found_test_files.size
+
+        KnapsackPro.logger.warn("Generating RSpec test examples JSON report to prepare your test suite to be split by test examples (by individual 'it'. Thanks to that a single test file can be split across parallel CI nodes). Analyzing #{test_files_count} test files.")
+
+        if test_files_count > 1000
+          KnapsackPro.logger.warn("You have more than 1000 test files, it may take longer to generate test examples. Please wait...")
+        end
+
+        # generate RSpec JSON report in separate process to not pollute RSpec state
+        cmd = 'bundle exec rake knapsack_pro:rspec_test_example_detector'
+        unless Kernel.system(cmd)
```

Comment:
@shadre I added it anyway. `detector.test_file_example_paths` method can raise when json file missing.